src/Maui/Prism.Maui/Common/MvvmHelpers.cs

```diff
@@ -303,6 +306,11 @@ public static int GetCurrentPageIndex(Page currentPage, IReadOnlyList<Page?> nav
                     return GetTarget(parentPage);
                 }
 
+                if (page.Parent is PrismWindow prismWindow)
+                {
+                    return GetTarget(prismWindow.Page);
+                }
```

Comment:
Since we are evaluating this 2x... let's switch this to a switch expression:

```cs
return page.Parent switch
{
    Page parentPage => GetTarget(parentPage),
    Window window => GetTarget(window.Page),
    _ => throw new InvalidOperationException(...)
}
```